internal/system/agent/services.go

```diff
@@ -34,6 +37,193 @@ const (
 	RESTART = "restart"
 )
 
+func InvokeMetrics(services []string, ctx context.Context) (MetricsRespMap, error) {
+
+	m := MetricsRespMap{}
+	m.Metrics = map[string]interface{}{}
+
+	// Loop through requested actions, along with (any) respectively-supplied parameters.
+	for _, service := range services {
+		LoggingClient.Info("invoke metrics")
+
+		if Configuration.MetricsMechanism == "direct-service" {
```

Comment:
this isn't quite how I imagined the abstraction would work. I think it should be organized like the service management, where we have a variable that is `interface{}` but might implement some other interfaces and we do a type cast to see if the variable implements any of the interfaces for gathering different kinds of metrics. For example, maybe the `direct-service` implementation only has CPU and Memory stats, so it only implements the interfaces `CPUStats` and `MemStats`, but the executor implementation also provides `NetworkIOStats` and `DiskIOStats` interfaces, so the executor implementation provides those stats as well. Again, I haven't been to many SMA WG meetings recently so I'm not sure how the design has progressed, but that's what I remember discussing.

Additionally, I think it's cleaner to have the inspection of the configuration value done during init, and assign the chosen implementation for the metrics gathering function, then this function doesn't need to read any information from the configuration struct.